vscode/src/debugger/session.ts

```diff
@@ -12,36 +12,85 @@ import {
   LoggingDebugSession,
   TerminatedEvent,
   logger,
+  Breakpoint,
+  StoppedEvent,
+  BreakpointEvent,
+  Thread,
+  StackFrame,
+  Source,
 } from "@vscode/debugadapter";
 
 import { FileAccessor } from "../common";
 import { DebugProtocol } from "@vscode/debugprotocol";
-import { ICompiler, log } from "qsharp";
+import {
+  BreakpointSpan,
+  ICompiler,
+  IDebugServiceWorker,
+  log,
+  Span,
+} from "qsharp";
 import { createDebugConsoleEventTarget } from "./output";
 import { ILaunchRequestArguments } from "./types";
 
 const ErrorProgramHasErrors = "program contains compile errors(s): cannot run.";
 const SimulationCompleted = "Q# simulation completed.";
 const ConfigurationDelayMS = 1000;
-const LaunchDelayMS = 100;
 
 function delay(ms: number): Promise<void> {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
 
 export class QscDebugSession extends LoggingDebugSession {
+  private static threadID = 1;
   private _fileAccessor: FileAccessor;
   private _compiler: ICompiler;
-
-  public constructor(fileAccessor: FileAccessor, compiler: ICompiler) {
+  private _debugger: IDebugServiceWorker;
+  private _config: vscode.DebugConfiguration;
+  private _breakpointLocations: Map<string, BreakpointSpan[]>;
+  private _breakpoints: Map<string, DebugProtocol.Breakpoint[]>;
+  private _failed: boolean;
+
+  public constructor(
+    fileAccessor: FileAccessor,
+    compiler: ICompiler,
+    debugService: IDebugServiceWorker,
+    config: vscode.DebugConfiguration
+  ) {
     super();
     this._fileAccessor = fileAccessor;
     this._compiler = compiler;
-
+    this._debugger = debugService;
```

Comment:
You can use [parameter properties](https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties) to initialize these to reduce verbosity.